### PR TITLE
ISL Conan 2.0 compatibility

### LINKED ISSUE #14915: [package] ISL/0.24 lacks support for Conan 2.0
### Description

Uses old-style imports

### Package and Environment Details

isl/0.24

### Conan profile

Any

### Steps to reproduce

conan-2.0 build -pr:b tools.jinja -pr:h tools.jinja --name isl --version 0.24 . 

### Logs

<details><summary>Click to expand log</summary>

```
conan-2.0 build -pr:b tools.jinja -pr:h tools.jinja --name isl --version 0.24 .

-------- Input profiles --------
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
[options]
cmake*:bootstrap=True
[tool_requires]
!openssl*: cmake/3.24.2

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
[options]
cmake*:bootstrap=True
[tool_requires]
!openssl*: cmake/3.24.2

ERROR: Error loading conanfile at '/tmp/tmp.UuQlL1yi0W/isl/conanfile.py': Unable to load conanfile in /tmp/tmp.UuQlL1yi0W/isl/conanfile.py
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmp/tmp.UuQlL1yi0W/isl/conanfile.py", line 1, in <module>
    from conans import AutoToolsBuildEnvironment, ConanFile, tools
ImportError: cannot import name 'AutoToolsBuildEnvironment' from 'conans' (/mathworks/devel/sandbox/marc/conan-2.0/conan-io/conans/__init__.py)
```

</details>

### PR DESCRIPTION
**Pull request by System-Arch**
Specify library name and version:  **isl/0.24**

Migrate to Conan 2.0. Resolves #14915

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/isl/all/conanfile.py [hunk 1]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
 
         os.unlink(os.path.join(os.path.join(self.package_folder, "lib", "libisl.la")))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "isl"
+        self.cpp_info.set_property("cmake_find_mode", "both")
+        self.cpp_info.set_property("cmake_file_name", "ISL")
+        self.cpp_info.set_property("cmake_target_name", "ISL::ISL")
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:124]:
```suggestion
        self.cpp_info.set_property("pkg_config_name", "isl")
```

there is no Find module or CMake config file upstream, do not dictate an unofficial name please

File: recipes/isl/all/conanfile.py [hunk 2]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
         self.cpp_info.libs = ["isl"]
+
+        self.cpp_info.names["pkg_config"] = "isl"
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:127]:
```suggestion
```

File: recipes/isl/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -1,8 +1,7 @@
 cmake_minimum_required(VERSION 3.1)
 project(test_package C)
 
-include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
-conan_basic_setup()
+find_package(ISL REQUIRED)
```
Comment (SpaceIm) [inline recipes/isl/all/test_package/CMakeLists.txt:4]:
```suggestion
find_package(isl REQUIRED CONFIG)
```

File: recipes/isl/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -1,8 +1,7 @@
...
 
 add_executable(${PROJECT_NAME} test_package.c)
-target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+target_link_libraries(${PROJECT_NAME} PRIVATE ISL::ISL)
```
Comment (SpaceIm) [inline recipes/isl/all/test_package/CMakeLists.txt:7]:
```suggestion
target_link_libraries(${PROJECT_NAME} PRIVATE isl::isl)
```

File: recipes/isl/all/test_package/conanfile.py [hunk 5]
```diff
@@ -1,17 +1,25 @@
...
+from conan import ConanFile
+from conan.tools.cmake import CMake, cmake_layout
+from conan.tools.build import cross_building
 import os
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    generators = "CMakeToolchain", "CMakeDeps", "VirtualRunEnv"
```
Comment (SpaceIm) [inline recipes/isl/all/test_package/conanfile.py:9]:
```suggestion
    generators = "CMakeToolchain", "CMakeDeps", "VirtualRunEnv"
    test_type = "explicit"
```

File: recipes/isl/all/test_v1_package/test_package.c [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#include "isl/ctx.h"
```
Comment (SpaceIm) [inline recipes/isl/all/test_v1_package/test_package.c:1]: do not duplicate code please, see recipe templates in conan-center doc

File: recipes/isl/all/conanfile.py [hunk 7]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
 
     def build(self):
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
+        # Support building with source from Git reop
+        with chdir(self, self.source_folder):
+            command = "./autogen.sh"
+            if os.path.exists(command) and not os.path.exists("configure"):
+                self.run(command)
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:107]: no sorry + it breaks if autotools are not installed on build machine

File: recipes/isl/all/conanfile.py [hunk 8]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
     def package(self):
-        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"), keep_path=False)
+        autotools = Autotools(self)
+        autotools.install()
 
         os.unlink(os.path.join(os.path.join(self.package_folder, "lib", "libisl.la")))
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:118]:
```suggestion
        rm(self, "*.la", os.path.join(os.path.join(self.package_folder, "lib"))
```

File: recipes/isl/all/conanfile.py [hunk 9]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
+        if self.settings.compiler == "msvc":
+            if Version(self.settings.compiler.version) >= 15:
+                tc.extra_cflags = ["-Zf"]
+            if Version(self.settings.compiler.version) >= 12:
+                tc.extra_cflags = ["-FS"]
+        env = tc.environment()
+        if self.settings.compiler == "msvc":
+            env.define("AR", f'{unix_path(self, self.dependencies["automake"].ar_lib)} lib')
+            env.define("CC", f'{unix_path(self, self.dependencies["automake"].compile)} cl -nologo {self.settings.compiler.runtime}')
+            env.define("CXX", f'{unix_path(self, self.dependencies["automake"].compile)} cl -nologo {self.settings.compiler.runtime}')
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:95]:
```suggestion
            env.define("CC", f'{unix_path(self, self.dependencies["automake"].compile)} cl -nologo')
            env.define("CXX", f'{unix_path(self, self.dependencies["automake"].compile)} cl -nologo')
```

File: recipes/isl/all/conanfile.py [hunk 10]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
-        return self._autotools
+            tc.configure_args.append("--with-gmp=system")
+            tc.configure_args.append(f'--with-gmp-prefix={unix_path(self, self.dependencies["gmp"].package_folder)}')
+        if self.settings.compiler == "msvc":
+            if Version(self.settings.compiler.version) >= 15:
+                tc.extra_cflags = ["-Zf"]
+            if Version(self.settings.compiler.version) >= 12:
+                tc.extra_cflags = ["-FS"]
+        env = tc.environment()
+        if self.settings.compiler == "msvc":
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:92]:
```suggestion
        if is_msvc(self):
```

File: recipes/isl/all/conanfile.py [hunk 11]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
-        if self.settings.compiler == "Visual Studio":
-            if tools.Version(self.settings.compiler.version) >= 15:
-                self._autotools.flags.append("-Zf")
-            if tools.Version(self.settings.compiler.version) >= 12:
-                self._autotools.flags.append("-FS")
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
+            tc.configure_args.append("--with-gmp=system")
+            tc.configure_args.append(f'--with-gmp-prefix={unix_path(self, self.dependencies["gmp"].package_folder)}')
+        if self.settings.compiler == "msvc":
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:86]:
```suggestion
        if is_msvc(self):
```

File: recipes/isl/all/conanfile.py [hunk 12]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
+            if Version(self.settings.compiler.version) >= 15:
+                tc.extra_cflags = ["-Zf"]
+            if Version(self.settings.compiler.version) >= 12:
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:89]: Visual Studio & msvc don't follow the same versioning in conan

File: recipes/isl/all/conanfile.py [hunk 13]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
-        ]
+        get(self, **self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self.source_folder)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append(f'--with-int={self.options.with_int}')
+        tc.configure_args.append("--enable-portable-binary")
+        tc.configure_args.append(f'--enable-shared={"yes" if self.options.shared else "no"}')
+        tc.configure_args.append(f'--enable-static={"yes" if not self.options.shared else "no"}')
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:82]:
```suggestion
```

comes for free in AutotoolsToolchain

File: recipes/isl/all/conanfile.py [hunk 14]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
-        if self.settings.compiler == "Visual Studio":
-            self.build_requires("automake/1.16.4")
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
+        if self.settings.compiler == "msvc":
+            self.tool_requires("automake/1.16.4")
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:68]:
```suggestion
        if is_msvc(self):
            self.tool_requires("automake/1.16.5")
```

File: recipes/isl/all/conanfile.py [hunk 15]
```diff
@@ -49,7 +47,7 @@ def validate(self):
         if self.options.with_int != "gmp":
             # FIXME: missing imath recipe
             raise ConanInvalidConfiguration("imath is not (yet) available on cci")
-        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 16 and self.settings.compiler.runtime == "MDd":
+        if self.settings.compiler == "msvc" and Version(self.settings.compiler.version) < 16 and self.settings.compiler.runtime == "MDd":
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:50]:
```suggestion
        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) < 16 and self.settings.compiler.runtime == "MDd":
```

File: recipes/isl/all/conanfile.py [hunk 16]
```diff
@@ -1,10 +1,13 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
-from contextlib import contextmanager
+from conan import ConanFile
+from conan.tools.files import chdir, copy, get, rmdir, apply_conandata_patches
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import unix_path
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:5]:
```suggestion
from conan.tools.microsoft import is_msvc, unix_path
```

File: recipes/isl/all/conanfile.py [hunk 17]
```diff
@@ -1,10 +1,13 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
-from contextlib import contextmanager
+from conan import ConanFile
+from conan.tools.files import chdir, copy, get, rmdir, apply_conandata_patches
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:2]:
```suggestion
from conan.tools.files import chdir, copy, get, rmdir
```

File: recipes/isl/all/conanfile.py [hunk 18]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
     def build(self):
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
+        # Support building with source from Git reop
+        with chdir(self, self.source_folder):
+            command = "./autogen.sh"
+            if os.path.exists(command) and not os.path.exists("configure"):
+                self.run(command)
+        apply_conandata_patches(self)
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:108]:
```suggestion
```

there is no patch

File: recipes/isl/all/conanfile.py [hunk 19]
```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
...
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make()
 
     def package(self):
-        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"), keep_path=False)
```
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:114]:
```suggestion
        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/isl/all/conanfile.py (hunk 7 above)
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:107]:
```suggestion
```

no sorry + it breaks if autotools are not installed on build machine

File: recipes/isl/all/conanfile.py (hunk 15 above)
Comment (SpaceIm) [inline recipes/isl/all/conanfile.py:50]: you have to write something like this instead:

```python
if msvc_runtime_flag(self) == "MDd":
    check_min_vs(self, "192")
```

File: recipes/isl/all/conanfile.py (hunk 17 above)
Comment (System-Arch) [reply recipes/isl/all/conanfile.py]: Back ported fix from 0.25 so now apply_conandata_patches is needed.

File: recipes/isl/all/conanfile.py (hunk 18 above)
Comment (System-Arch) [reply recipes/isl/all/conanfile.py]: Back ported fix from 0.25 so now apply_conandata_patches is needed.

File: recipes/isl/all/conanfile.py (hunk 7 above)
Comment (System-Arch) [reply recipes/isl/all/conanfile.py]: Hi @SpaceIm,
The requisite autotools are specified as tool_requires so they don't need to be installed on the build machine

File: recipes/isl/all/conanfile.py (hunk 15 above)
Comment (System-Arch) [reply recipes/isl/all/conanfile.py]: I believe this is addressed with the change summited below

File: recipes/isl/all/conanfile.py (hunk 19 above)
Comment (System-Arch) [reply recipes/isl/all/conanfile.py]: Fixed.

File: recipes/isl/all/conanfile.py (hunk 16 above)
Comment (System-Arch) [reply recipes/isl/all/conanfile.py]: Fixed

File: recipes/isl/all/test_package/conanfile.py (hunk 5 above)
Comment (System-Arch) [reply recipes/isl/all/test_package/conanfile.py:9]: Fixed

File: recipes/isl/all/test_package/CMakeLists.txt (hunk 4 above)
Comment (System-Arch) [reply recipes/isl/all/test_package/CMakeLists.txt]: Fixed

File: recipes/isl/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (System-Arch) [reply recipes/isl/all/test_package/CMakeLists.txt]: Fixed

File: recipes/isl/all/conanfile.py (hunk 2 above)
Comment (System-Arch) [reply recipes/isl/all/conanfile.py]: Fixed

File: recipes/isl/all/conanfile.py (hunk 1 above)
Comment (System-Arch) [reply recipes/isl/all/conanfile.py]: Fixed

File: recipes/isl/all/conanfile.py [hunk 20]
```diff
@@ -62,72 +66,71 @@ def _settings_build(self):
...
+                tc.extra_cflags = ["-Zf"]
+            if compiler_version >= "180":
+                tc.extra_cflags = ["-FS"]
+        # Visual Studio support for Conan 1.x; Can be remvoed when 2.0 is default
+        if self.settings.get_safe("compiler") == "Visual Studio":
+            compiler_version = self.settings.get_safe("compiler.version")
+            if compiler_version >= "15":
+                tc.extra_cflags = ["-Zf"]
+            if compiler_version >= "12":
+                tc.extra_cflags = ["-FS"]
```
Comment (valgur) [inline recipes/isl/all/conanfile.py:104]:
```suggestion
        if is_msvc(self):
            compiler_version = self.settings.get_safe("compiler.version")
            if compiler_version >= "191":
                tc.extra_cflags += ["-Zf"]
            if compiler_version >= "180":
                tc.extra_cflags += ["-FS"]
        # Visual Studio support for Conan 1.x; Can be remvoed when 2.0 is default
        if self.settings.get_safe("compiler") == "Visual Studio":
            compiler_version = self.settings.get_safe("compiler.version")
            if compiler_version >= "15":
                tc.extra_cflags += ["-Zf"]
            if compiler_version >= "12":
                tc.extra_cflags += ["-FS"]
```

**Review by valgur — CHANGES_REQUESTED** [review]

**Comment by RubenRBS** [comment]
Please bot do not close it

**Comment by AndreyMlashkin** [comment]
@System-Arch, could you merge @valgur contribution to your branch?
Those changes are importatnt

**Pull request closed without merging**